### PR TITLE
Fix gremlin endpoint change and error handling, fixes #322

### LINKED ISSUE #322: Keep 'Query...' more than 20 mins after excuting g.V()
Repro steps:

1. Open one graph.
2. Input g.V() in query testbox of this graph explorer.
3. Click 'Excute'.

Expect:
Under Graph, you should see the following vertices and edges.

Actual:
Keep 'Query...' under 'Graph/JSON' more than 20 mins after excuting g.V().

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Third time's the charm...

### CODE REVIEW

File: src/graph/GraphViewServer.ts [hunk 1]
```diff
@@ -332,14 +362,26 @@ export class GraphViewServer extends EventEmitter {
...
+      streamError = err;
+    }
+
     return new Promise<[{}[]]>((resolve, reject) => {
       client.execute(gremlinQuery, {}, (err, results) => {
-        if (err) {
+        if (streamError) {
+          this.log("Gremlin client error: ", streamError.message || streamError.toString());
+          reject(streamError);
+        } else if (err) {
```
Comment (EricJizbaMSFT) [inline src/graph/GraphViewServer.ts]: Not quite sure what the difference between `streamError` and `err` is. Could you give an example of each?

File: src/graph/GraphViewServer.ts [hunk 2]
```diff
@@ -332,14 +362,26 @@ export class GraphViewServer extends EventEmitter {
...
+    function handleError(err) {
+      console.warn(err);
+      streamError = err;
+    }
+
     return new Promise<[{}[]]>((resolve, reject) => {
       client.execute(gremlinQuery, {}, (err, results) => {
-        if (err) {
+        if (streamError) {
+          this.log("Gremlin client error: ", streamError.message || streamError.toString());
```
[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply src/graph/GraphViewServer.ts]: Yes, that is the intention eventually. Perhaps something that goes off of a VSCode setting?  Logging can be very helpful in certain scenarios (including being able to ask customers to turn it on).

File: src/graph/GraphViewServer.ts (hunk 1 above)
Comment (StephenWeatherford) [reply src/graph/GraphViewServer.ts]: Essentially streamError is anything that the gremlin client is handling itself, i.e. errors from websocket/etc.  I'll change the name to that.  Personally I think they should be routing everything the same way.

File: src/graph/gremlinEndpoints.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import { IAzureTreeItem } from 'vscode-azureextensionui';
+import CosmosDBManagementClient = require("azure-arm-cosmosdb");
+
+export interface IGremlinEndpoint {
+    host: string;
+    port: number;
+    ssl: boolean;
+}
+
+export async function TryGetGremlinEndpointFromAzure(client: CosmosDBManagementClient, documentEndpoint: string, resourceGroup: string, account: string): Promise<IGremlinEndpoint | undefined> {
```
[the remark this answers is not in the document]
Comment (StephenWeatherford) [reply src/graph/gremlinEndpoints.ts]: Thanks, it used to be.  Fixed.

**Pull request merged**